feat(Wolfy): improvements, add buttons to view profile and join game

Tenebrosful opened the pull request:
**Describe the changes in this pull request:**
- Update with new html class (was broken before)
- Add a button to go to user's profile
- Handle when viewing leaderboard
- Handle when viewing skin inventory
- Improve settings and shop text
- Add a button to join the game when you are in lobby (can be disable in settings)

<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>
<!-- Required when a presence has been added or modified --->
<!-- Attach screenshot(s) here --->
![image](https://user-images.githubusercontent.com/48710891/163489218-49b708d4-7e63-4eea-88a3-a46b56fe1257.png)

![image](https://user-images.githubusercontent.com/48710891/163489246-5a507b74-f02e-47e0-8c8f-867a14a4aa24.png)

![image](https://user-images.githubusercontent.com/48710891/163489280-dc304d4b-3a34-4227-af58-e24336d591ff.png)

![image](https://user-images.githubusercontent.com/48710891/163489299-a2446541-9fbc-4b0a-9e72-438676d353ff.png)

![image](https://user-images.githubusercontent.com/48710891/163489317-14afd81c-ac3f-4490-bca2-4d83d739010d.png)

![image](https://user-images.githubusercontent.com/48710891/163489348-749e028a-bd50-4e9d-971f-cc2683a6b0aa.png)

![image](https://user-images.githubusercontent.com/48710891/163489372-60e06b9c-4b33-4143-a85a-b127715ef36f.png)


</details>

Last commits (15/04/2022)

- Improve Visit Profil button (Add the username)
- Add Visit Profil button when looking at a player on leaderboard
- JoinGame button is now configurable

Note: I just noticed that is you use an anonymous username, it will be used in the Visit Profile Button, that's why i made it configurable

<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>
<!-- Required when a presence has been added or modified --->
<!-- Attach screenshot(s) here --->

![image](https://user-images.githubusercontent.com/48710891/163580120-cc1cd0b4-9898-460c-a05b-8844a8590d3d.png)

![image](https://user-images.githubusercontent.com/48710891/163580257-829f542b-1a74-4b1b-89db-f6363aff7889.png)

</details>

Last commits (19/04/2022)

- Improve text when reading articles
- Remove unused getStrings
- Change clientID to Official Wolfy Discord App (owned by Wolfy's Admin, that will make it easier to add and change images)
- Invert 1st and 2nd line when looking leaderboard
- Handle help.wolfy.fr
- Add a "Visit Wolfy Website" button if there is less than 2 buttons used
- Refactor add buttons functions
- Refactor if condition when split path

<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>
<!-- Required when a presence has been added or modified --->
<!-- Attach screenshot(s) here --->

Soon

</details>

Commit:
fix(fix class name to fit new interface)

Commit:
feat(Add visit profile button when playing)

Commit:
Add a button to visit user profile

Commit:
Handle leaderboard looking

Commit:
Add skin inventory looking

Commit:
Improve settings and shop text

Commit:
Add a configurable buttonto join the game

Commit:
Add myself to contributors

Commit:
Update version

Tenebrosful commented:
Was my first time with PreMid developpement, if it's great, i will probably look to improve some things like making it multi-language

EncryptedDev commented:
@Kozou4ever will check the strings 👍🏼

Kozou4ever commented:
LGTM

Commit:
Add doc about anonymous username in game

Commit:
Improve Visit Profile Button

Commit:
Configurable joinGameButton

Tenebrosful commented:
Last commits

- Improve Visit Profil button (Add the username)
- Add Visit Profil button when looking at a player on leaderboard
- JoinGame button is now configurable

Note: I just noticed that is you use an anonymous username, it will be used in the Visit Profile Button, that's why i made it configurable

<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>
<!-- Required when a presence has been added or modified --->
<!-- Attach screenshot(s) here --->
![image](https://user-images.githubusercontent.com/48710891/163580120-cc1cd0b4-9898-460c-a05b-8844a8590d3d.png)
![image](https://user-images.githubusercontent.com/48710891/163580257-829f542b-1a74-4b1b-89db-f6363aff7889.png)

</details>

Commit:
Fix missing await

Tenebrosful committed:
Merge branch 'main' into main

Slowlife01 commented on websites/W/Wolfy/presence.ts:
```diff
@@ -27,6 +26,85 @@ function getTimestamps(audioTime: number, audioDuration: string): number[] {
+	presenceData: PresenceData,
+	username: string
+) {
+	if (!(await presence.getSetting("visitProfileButton"))) return;
+	if (!presenceData.buttons) {
+		presenceData.buttons = [
+			{
+				label: `Visiter le profil de ${username}`,
+				url: `https://wolfy.fr/leaderboard/${username}`
+			}
+		];
+	} else {
+		presenceData.buttons[1] = {
+			label: `Visiter le profil de ${username}`,
+			url: `https://wolfy.fr/leaderboard/${username}`
+		};
+	}
```
You can only have 32 characters on the labels
So this may not be a good idea

Tenebrosful replied:
Thx, i just commited a fix about that.

If the label is longer than 32 characters i cut it now.

Commit:
Prevent > 32 characters label

https://github.com/PreMiD/Presences/pull/6083#discussion_r852644338

Commit:
Lint

Slowlife01 commented on websites/W/Wolfy/presence.ts:
```diff
@@ -27,6 +26,89 @@ function getTimestamps(audioTime: number, audioDuration: string): number[] {
+	if (!(await presence.getSetting("visitProfileButton"))) return;
+
+	let label = `Visiter le profil de ${username}`;
+	if (label.length > 32) label = `${label.slice(0, 31)}…`;
```
31 + 3 = 34

Tenebrosful replied:
It's not "..." it's "…" that is a single unicode character

Tenebrosful replied:
![image](https://user-images.githubusercontent.com/48710891/163954025-6f08d949-2502-4beb-b172-a0eab4dae421.png)

Slowlife01 commented on websites/W/Wolfy/presence.ts:
```diff
@@ -2,7 +2,6 @@ const presence = new Presence({
 		clientId: "721798189111443537"
 	}),
 	strings = presence.getStrings({
-		play: "presence.playback.playing",
 		browsing: "presence.activity.browsing"
 	});
```
Do we really need this?

Tenebrosful replied:
It was from old code, I removed it because `play` is translated in `lecture` like if it was "playing music",  it doesn't fit in french

Slowlife01 replied:
It seems pretty silly that only 1 string is used, considering that all the other ones are hardcoded.

Tenebrosful replied:
I will look at multi-langage and adding more strings in this to have no more hardcoded strings

Tenebrosful replied:
According to Guidelines, i can't create custom strings with multiLanguage until this presence have 1000 users , so i will just hardcoded all strings

Commit:
Add specific text when reading article

Commit:
Remove getStrings

Commit:
Add image when looking at the leaderboard

Commit:
Remove default image reading in fallback

Commit:
Change clientId to Official Wolfy Discord App

Commit:
Add comment about clientId

Commit:
Invert 1st (details) and 2nd (state) line

Commit:
Improve text when reading and browsing articles

Commit:
Handle help.wolfy.fr

Commit:
Add Visit website button if there is less than 2 buttons

Commit:
Lint

Commit:
Rework add buttons functions

Commit:
Fix array index

Commit:
Fix func rename

Commit:
Fix condition path.split

Commit:
Handle category help center

Commit:
Add missing consult category button

Tenebrosful commented:
Last commits

- Improve text when reading articles
- Remove unused getStrings
- Change clientID to Official Wolfy Discord App (owned by Wolfy's Admin, that will make it easier to add and change images)
- Invert 1st and 2nd line when looking leaderboard
- Handle help.wolfy.fr
- Add a "Visit Wolfy Website" button if there is less than 2 buttons used
- Refactor add buttons functions
- Refactor if condition when split path

Note: I just noticed that is you use an anonymous username, it will be used in the Visit Profile Button, that's why i made it configurable

<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>
<!-- Required when a presence has been added or modified --->
<!-- Attach screenshot(s) here --->

Soon

</details>

Commit:
Update smallImage with new application

Slowlife01 commented on websites/W/Wolfy/presence.ts:
```diff
@@ -1,10 +1,7 @@
-		play: "presence.playback.playing",
-		browsing: "presence.activity.browsing"
-	});
+	clientId: "501842028569559061" // Official Wolfy Discord App Client ID, owned by Wolfy's Admin
```
Have you asked the author about this?

Tenebrosful replied:
hmmm, that's something that i completely forgot because it's was just a presence request at first #286.

@fr0r is the original author, i can't contact him in DMs on Discord about that, i will try to ping on the PreMid server since he isn't on Wolfy's Discord

Tenebrosful committed:
Merge branch 'main' into main

Tenebrosful committed:
Anticipate domainname change

https://wolfy.fr/ will become https://wolfy.net/ the 01/05/2022

Tenebrosful committed:
Update metadata.json

Tenebrosful committed:
Update metadata.json

Tenebrosful committed:
Transfert ownership

https://discord.com/channels/493130730549805057/554447633197039627/966721156973469766

Commit:
fix(don't show visit profil button if username is undefined)

Commit:
fix joinGameButton missing await

Tenebrosful committed:
Merge branch 'main' into main

Tenebrosful committed:
Merge branch 'main' into main
